fixes for HTML parser failing

kdutia opened the pull request:
Fixes to the HTML parser from examining [loads of HTML pages for which we got no text](https://docs.google.com/spreadsheets/d/1W7OeyNcZd3atXwTbS3XKbqoknFqAEvhEDslN0mxYSpA/edit?gid=566509493#gid=566509493). This should fix all of the issues I saw on domains that had issues several times – once this is deployed, I'll run all of the docs through it and look at any outstanding failures.

Some of this will hopefully soon be replaced by converting htmls to pdf (if the spike is successful), but some of the HTML-getting tricks here will be needed for that anyway!

On fix per commit, if those are easier to look at:

- fix pydantic validationerror: don't try to detect html title and date. fixes URL https://ap.fftc.org.tw/article/1393
- fix JSON serialization error for `AnyHTTPURL`. fixes URL https://busquedas.elperuano.pe/download/url/decreto-supremo-que-declara-de-interes-nacional-la-emergenci-decreto-supremo-n-003-2022-minam-2033317-1
- playwright waits for network to be idle before getting HTML. fixes URL https://diariodarepublica.pt/dr/legislacao-consolidada/portaria/2021-158240165
- playwright uses realistic user-agent header. fixes https://www.legifrance.gouv.fr/ domains

## A note on flaky translation tests

The tests failed once as an extra translated document was appearing for an English language HTML. I couldn't reproduce these results locally so I added some better logging around the code which seemed to be causing issues, then the tests passed.

So, that's the reason for that logging change – and I guess a heads up about the translation function!

THOR300 commented on src/html_parser/newsplease.py:
```diff
@@ -140,8 +140,6 @@ def _newsplease_article_to_parsed_html(
             document_content_type=input.document_content_type,
             document_slug=input.document_slug,
             html_data=HTMLData(
-                detected_title=newsplease_article.title,
```
Just confirmed by searching the cpr codebase in github and I don't think these are used anywhere.

kdutia replied:
yeah, they aren't! this was me being ambitious when originally writing this and wondering whether we could automatically set document titles using detected titles from HTMLs

THOR300 reviewed (approved):
Approving to unblock, question as to why the `poetry.lock` has updated though?

kdutia commented:
> Approving to unblock, question as to why the poetry.lock has updated though?

i added and removed a package and it updated a few things that looked harmless, seemingly related to package constraints it found on the internet. i can revert if you'd like?
